corehq/apps/export/export.py

```diff
@@ -0,0 +1,105 @@
+import contextlib
+import os
+import tempfile
+
+from corehq.apps.es import FormES, CaseES
+from corehq.apps.export.models.new import (
+    CaseExportInstance,
+    FormExportInstance,
+)
+from couchexport.export import FormattedRow, get_writer
+
+
+class _Writer(object):
+    """
+    An object that provides a friendlier interface to couchexport.ExportWriters.
+    """
+    def __init__(self, writer):
+        # An instance of a couchexport.ExportWriter
+        self.writer = writer
+
+    @contextlib.contextmanager
+    def open(self, tables):
+        """
+        Open the _Writer for writing. This must be called before using _Writer.write()
+        Note that this function returns a context manager!
+        """
+
+        # Create and open a temp file
+        fd, path = tempfile.mkstemp()
+        with os.fdopen(fd, 'wb') as file:
+
+            # open the ExportWriter
+            headers = [(t.identifier, (t.get_headers(),)) for t in tables]
+            table_titles = {t.identifier: t.name for t in tables}
+            self.writer.open(headers, file, table_titles=table_titles)
+            yield
+
+        self.writer.close()
+
+    def write(self, table, row):
+        return self.writer.write([(table.identifier, [FormattedRow(data=row.data)])])
+
+    def get_preview(self):
+        return self.writer.get_preview()
+
+
+def get_export_file(export_instance, filters):
+    """
+    Return an export file for the given ExportInstance and list of filters
+    """
+    docs = _get_export_documents(export_instance, filters)
+    return _write_export_file(export_instance, docs)
+
+
+def _get_export_documents(export_instance, filters):
+    # TODO: This function will be different for couch
+    query = _get_base_query(export_instance)
+    for filter in filters:
+        query = query.filter(filter.to_es_filter())
+    result = query.run()
+    return result.hits
```

Comment:
Ah ha, I didn't know about scroll and scan, thanks! I added sorting to the base queries when I had been thinking about paginated exports, where the queries for each page could be run several minutes apart (to enforce a rate limit). Since we're tabling paginated exports for now, I could remove this requirement, although it might be useful to keep it so that the transition to paginated exports will be a little smoother.

I'll just implement scroll for now